napari/plugins/_tests/conftest.py

```diff
@@ -23,3 +27,40 @@ def builtin_plugin_manager(plugin_manager):
             plugin_manager.unregister(mod)
     assert plugin_manager.get_plugins() == set([napari.plugins._builtins])
     return plugin_manager
+
+
+@pytest.fixture
+def temporary_hookimpl(plugin_manager):
+    """A fixture that can be used to insert a HookImpl in the hook call loop.
+
+    Example
+    -------
+
+    .. code-block: python
+
+        def bad_write_points(path, data, meta):
+            raise ValueError("shoot!")
+
+        with temporary_hookimpl(bad_write_points, 'napari_write_points'):
+            with pytest.raises(PluginCallError):
+                writer(tmpdir, layer_data, plugin_manager)
+    """
+
+    @contextmanager
+    def inner(
+        func, specname, tryfirst=True, trylast=None, plugin_name="<temp>"
+    ):
+        caller = getattr(plugin_manager.hook, specname)
+        HookimplMarker('napari')(tryfirst=tryfirst, trylast=trylast)(func)
```

Comment:
tbh this initially confused me because I was wondering why you would create a class without assigning it to anything

Reply:
I can leave a comment if it would help?  `HookimplMarker` creates a decorator.  And when we call it on `func` it just sets an attribute on `func`